main/src/com/google/refine/operations/column/ColumnAdditionByFetchingURLsOperation.java

```diff
@@ -213,66 +217,89 @@ public void run() {
                 _project.processManager.onFailedProcess(this, new Exception("Another column already named " + _newColumnName));
                 return;
             }
-            
+
             List<CellAtRow> urls = new ArrayList<CellAtRow>(_project.rows.size());
-            
+
             FilteredRows filteredRows = _engine.getAllFilteredRows();
             filteredRows.accept(_project, createRowVisitor(urls));
-            
+
             List<CellAtRow> responseBodies = new ArrayList<CellAtRow>(urls.size());
             for (int i = 0; i < urls.size(); i++) {
                 CellAtRow urlData = urls.get(i);
-                long start = System.currentTimeMillis();
-                CellAtRow cellAtRow = fetch(urlData);
+                CellAtRow cellAtRow = cachedFetch(urlData);
                 if (cellAtRow != null) {
                     responseBodies.add(cellAtRow);
                 }
-                
+
                 _progress = i * 100 / urls.size();
-                try {
-                    long delay = _delay - (System.currentTimeMillis() - start);
-                    if (delay > 0) {
-                        Thread.sleep(delay);
-                    }
-                } catch (InterruptedException e) {
-                    if (_canceled) {
-                        break;
-                    }
-                }
+
+                if (_canceled) {
+                    break;
+	        }
             }
-            
+
+	    _urlCache.clear();
```

Comment:
clear() won't shrink the HashMap.
[WeakHashMap](https://docs.oracle.com/javase/8/docs/api/java/util/WeakHashMap.html) maybe better in this case. 
An entry in a WeakHashMap will automatically be removed when its key is no longer in ordinary use.